File: app/src/main/java/com/akay/fitnass/data/RunsRepositoryImpl.java
package com.akay.fitnass.data;

import android.arch.lifecycle.LiveData;

import com.akay.fitnass.data.db.dao.ActiveRunsDao;
import com.akay.fitnass.data.db.dao.RunsDao;
import com.akay.fitnass.data.model.ActiveRuns;
import com.akay.fitnass.data.model.Runs;

import java.util.List;

public class RunsRepositoryImpl implements RunsRepository {
    private final ActiveRunsDao mActiveRunsDao;
    private final RunsDao mRunsDao;
    private final LiveData<ActiveRuns> mActiveRuns;
    private final LiveData<List<Runs>> mRunsList;

    public RunsRepositoryImpl(ActiveRunsDao activeRunsDao, RunsDao runsDao) {
        mActiveRunsDao = activeRunsDao;
        mRunsDao = runsDao;

        mActiveRuns = mActiveRunsDao.getLiveActiveRuns();
        mRunsList = mRunsDao.getAllLive();
    }

    @Override
    public LiveData<ActiveRuns> getActiveRuns() {
        return mActiveRuns;
    }

    @Override
    public LiveData<List<Runs>> getRuns() {
        return mRunsList;
    }

    @Override
    public Runs getById(long id) {
        return mRunsDao.getById(id);
    }

    @Override
    public void upsertActiveRuns(ActiveRuns activeRuns) {
        activeRuns.setId(ActiveRuns.ID);
        mActiveRunsDao.upsert(activeRuns);
    }

    @Override
    public void deleteActiveRuns() {
        mActiveRunsDao.deleteActiveRuns();
    }

    @Override
    public void saveRuns(Runs runs) {
        mRunsDao.insert(runs);
    }
}

File: app/src/main/java/com/akay/fitnass/data/RunsRepositoryImpl.kt
package com.akay.fitnass.data

import android.arch.lifecycle.LiveData

import com.akay.fitnass.data.db.dao.ActiveRunsDao
import com.akay.fitnass.data.db.dao.RunsDao
import com.akay.fitnass.data.model.ActiveRuns
import com.akay.fitnass.data.model.Runs

class RunsRepositoryImpl(private val mActiveRunsDao: ActiveRunsDao, private val mRunsDao: RunsDao) : RunsRepository {

    private val mActiveRuns: LiveData<ActiveRuns> = mActiveRunsDao.getLiveActiveRuns()
    private val mRunsList: LiveData<List<Runs>> = mRunsDao.getAllLive()

    override fun getActiveRuns(): ActiveRuns {
        return mActiveRunsDao.getActiveRuns()
    }

    override fun getLiveActiveRuns(): LiveData<ActiveRuns> = mActiveRuns

    override fun getLiveRuns(): LiveData<List<Runs>> = mRunsList

    override fun getById(id: Long): Runs = mRunsDao.getById(id)

    override fun upsertActiveRuns(activeRuns: ActiveRuns) {
        activeRuns.id = ActiveRuns.ID
        mActiveRunsDao.insert(activeRuns)
    }

    override fun deleteActiveRuns() = mActiveRunsDao.deleteActiveRuns()

    override fun saveRuns(runs: Runs) = mRunsDao.insert(runs)

    override fun deleteRuns(runs: Runs) = mRunsDao.delete(runs)
}
